### PR TITLE
ECDC-2501: Add unittest for LokiScriptModel

### PR DESCRIPTION
**Pull request by ebadkamil**

### CODE REVIEW

File: test/nicos_ess/loki/test_loki_scriptbuilder_model.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+
+def test_row_inserted_at_position():
+    data = [
+        ['00', '01', '02'],
+        ['10', '11', '12'],
+        ['20', '21', '22'],
+        ['30', '31', '32'],
+    ]
+    # Create table with DATA
```
Comment (mattclarke) [inline test/nicos_ess/loki/test_loki_scriptbuilder_model.py:53]:
```suggestion
    # Create table with data
```
Why are you shouting? ;)

File: test/nicos_ess/loki/test_loki_scriptbuilder_model.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    model = LokiScriptModel(HEADERS, num_rows)
+    if data is not None:
+        model.table_data = data
+    return model
+
+
+def test_initialization_done_correctly():
+    model = create_loki_script_model()
+    # check if initialized with empty data
+    assert not any(data for data in sum(model.table_data, []))
```
Comment (mattclarke) [inline test/nicos_ess/loki/test_loki_scriptbuilder_model.py:43]: Check dimensions too

File: test/nicos_ess/loki/test_loki_scriptbuilder_model.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+
+def test_row_inserted_at_position():
```
Comment (mattclarke) [inline test/nicos_ess/loki/test_loki_scriptbuilder_model.py:46]:
```suggestion
def test_inserting_empty_row():
```

File: test/nicos_ess/loki/test_loki_scriptbuilder_model.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+    assert model.table_data == [
+        ['A', 'B', ''],
+        ['C', 'D', ''],
+        ['', '', ''],
+        ['', '', ''],
+    ]
+
+
+def test_clipboard_data_gets_pasted_in_empty_table_at_top_right():
```
Comment (mattclarke) [inline test/nicos_ess/loki/test_loki_scriptbuilder_model.py:112]: Name should be something about data outside the columns being ignored

File: test/nicos_ess/loki/test_loki_scriptbuilder_model.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+
+    assert model.table_data == [
+        ['', '', 'A'],
+        ['', '', 'C'],
+        ['', '', ''],
+        ['', '', ''],
+    ]
+
+
+def test_clipboard_data_gets_pasted_in_empty_table_at_bottom_left():
```
Comment (mattclarke) [inline test/nicos_ess/loki/test_loki_scriptbuilder_model.py:127]: Again, something about table being expanded when pasting at the bottom

File: test/nicos_ess/loki/test_loki_scriptbuilder_model.py [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+    assert model.table_data == [
+        ['00', '01', '02'],
+        ['A', 'B', '12'],
+        ['C', 'D', '22'],
+        ['30', '31', '32'],
+    ]
+
+
+def test_clipboard_data_gets_pasted_in_table_with_one_hidden_column():
+    model = create_loki_script_model()
```
Comment (mattclarke) [inline test/nicos_ess/loki/test_loki_scriptbuilder_model.py:169]: hidden columns skipped when pasting data

**Comment by mattclarke** [comment]
Looks great, just some minor comments about naming.

When naming tests, sometimes (but not always) it helps to think as them as documentation that is telling other developers about the behaviour of the system being tested.

**Comment by ebadkamil** [comment]
Done.

File: test/nicos_ess/loki/test_loki_scriptbuilder_model.py [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+
+def test_inserting_empty_row():
+    data = [
+        ['00', '01', '02'],
+        ['10', '11', '12'],
+        ['20', '21', '22'],
+        ['30', '31', '32'],
+    ]
+    # Create table with DATA
```
Comment (mattclarke) [inline test/nicos_ess/loki/test_loki_scriptbuilder_model.py:57]:
```suggestion
    # Create table with data
```

File: test/nicos_ess/loki/test_loki_scriptbuilder_model.py [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+
+
+def test_inserting_empty_row():
+    data = [
+        ['00', '01', '02'],
+        ['10', '11', '12'],
+        ['20', '21', '22'],
+        ['30', '31', '32'],
+    ]
+    # Create table with data
```
Comment (mattclarke) [inline test/nicos_ess/loki/test_loki_scriptbuilder_model.py:57]:
```suggestion
    # Create model with data
```

File: test/nicos_ess/loki/test_loki_scriptbuilder_model.py [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+
+def test_removing_rows():
+    data = [
+        ['00', '01', '02'],
+        ['10', '11', '12'],
+        ['20', '21', '22'],
+        ['30', '31', '32'],
+    ]
+    # Create table with data
```
Comment (mattclarke) [inline test/nicos_ess/loki/test_loki_scriptbuilder_model.py:76]:
```suggestion
    # Create model with data
```

File: test/nicos_ess/loki/test_loki_scriptbuilder_model.py [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+
+
+def test_clipboard_data_gets_pasted_at_index_in_table_with_data():
+    data = [
+        ['00', '01', '02'],
+        ['10', '11', '12'],
+        ['20', '21', '22'],
+        ['30', '31', '32'],
+    ]
+    # Create table with data
```
Comment (mattclarke) [inline test/nicos_ess/loki/test_loki_scriptbuilder_model.py:157]:
```suggestion
    # Create model with data
```

**Pull request merged**